ui/components/component-library/avatar-token/avatar-token.test.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable jest/require-top-level-describe */
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { AvatarToken } from './avatar-token';
+
+describe('AvatarToken', () => {
+  const args = {
+    tokenName: 'ast',
+    tokenImageUrl: './AST.png',
+    showHalo: false,
+  };
+
+  it('should render correctly', () => {
+    const { getByTestId } = render(<AvatarToken data-testid="avatar-token" />);
+    expect(getByTestId('avatar-token')).toBeDefined();
+  });
+
+  it('should render image Avatar', () => {
+    render(<AvatarToken {...args} />);
+    const image = screen.getByRole('img');
+    expect(image).toBeDefined();
+    expect(image).toHaveAttribute('src', args.tokenImageUrl);
+  });
+
+  it('should render Avatar Name Initial if no tokenImageUrl is provided', () => {
```

Comment:
Until we break the component out into it's own component called AvatarInitial lets just call it the first letter of the tokenName

```suggestion
  it('should render the first letter of the tokenName prop if no tokenImageUrl is provided', () => {
```